bdk_rpc_example/src/main.rs

```diff
@@ -0,0 +1,182 @@
+        RpcCommands::Scan {
+            initial_height,
+            stop_gap,
+        } => {
+            let local_chain = keychain_tracker.chain().checkpoints().clone();
+
+            let (chan, recv) = channel::<RpcData>();
+            let chan_term = chan.clone();
+
+            // ctrl+C handler
+            ctrlc::set_handler(move || match chan_term.send(RpcData::Stop(false)) {
+                Ok(_) => log::info!("caught SIGTERM"),
+                Err(err) => log::warn!("failed to send SIGTERM: {}", err),
+            })
+            .expect("failed to set Ctrl+C handler");
+
+            // emit blocks thread
+            let join_handle =
+                std::thread::spawn(move || client.emit_blocks(&chan, &local_chain, initial_height));
+
+            for data in recv.iter() {
+                let mut update = ChainGraph::<TxHeight>::default();
+                let mut last_active_indexes = BTreeMap::new();
+
+                let txs = match data {
+                    RpcData::Blocks(blocks) => {
+                        for (&height, block) in blocks.iter() {
+                            update.insert_checkpoint(BlockId {
+                                height: height as _,
+                                hash: block.block_hash(),
+                            })?;
+                        }
+                        blocks
+                            .into_iter()
+                            .flat_map(|(height, block)| {
+                                block
+                                    .txdata
+                                    .into_iter()
+                                    .map(move |tx| (TxHeight::Confirmed(height as _), tx))
+                            })
+                            .collect::<Vec<_>>()
+                    }
+                    RpcData::Mempool(txs) => txs
+                        .into_iter()
+                        .map(|tx| (TxHeight::Unconfirmed, tx))
+                        .collect::<Vec<_>>(),
+                    RpcData::Stop(finished) => {
+                        log::info!("terminating... sync_finished={}", finished);
+                        drop(recv);
+                        break;
+                    }
+                };
+
+                for (height, tx) in txs {
+                    ensure_spk_cached(&mut keychain_tracker.txout_index, stop_gap);
+
+                    if keychain_tracker.txout_index.is_relevant(&tx) {
+                        update.insert_tx(tx.clone(), Some(height))?;
+                    }
+
+                    for (keychain, this_ind) in keychain_tracker.txout_index.scan(&tx) {
+                        last_active_indexes
+                            .entry(keychain)
+                            .and_modify(|ind| {
+                                if this_ind > *ind {
+                                    *ind = this_ind;
+                                }
+                            })
+                            .or_insert(this_ind);
```

Comment:
Like I've been saying I think you should just have a method on `txout_index` to get this value. Very useful method in general that just happens to be missing.